queries.sql

```diff
@@ -39,3 +39,88 @@ SELECT * FROM animals a INNER JOIN owners o ON a.owner_id = o.id WHERE o.full_na
 SELECT * FROM animals a INNER JOIN owners o ON a.owner_id = o.id WHERE o.full_name ='Dean Winchester' AND a.escape_attempts <= 0;
 
 SELECT o.full_name, COUNT(*) FROM owners o LEFT JOIN animals a ON o.id =  a.owner_id GROUP BY o.full_name ORDER BY COUNT DESC LIMIT 1;
+
+SELECT v.name AS vet_name, a.name AS animal_name, vs.date_of_visit
+  FROM visits vs INNER JOIN animals a ON a.id = vs.animals_id
+  INNER JOIN vets v ON v.id = vs.vets_id
+  WHERE v.name = 'William Tatcher'
+  ORDER BY vs.date_of_visit DESC
+  LIMIT 1;
+
+SELECT v.name as vet_name, COUNT(date_of_visit) 
+  FROM visits vs LEFT JOIN vets v ON vs.vets_id = v.id
+  WHERE name = 'Stephanie Mendez'
+  GROUP BY v.name;
+
+SELECT 
+  sp.id AS specialty_id, 
+  sp.species_id, 
+  sp.vets_id, 
+  v.name AS vet_name, 
+  s.name AS species_name  
+  FROM specializations sp FULL OUTER JOIN species s ON s.id = sp.species_id
+  FULL OUTER JOIN vets v 
+    ON v.id = sp.vets_id;
+
+SELECT 
+  a.name, 
+  v.name AS vet_name, 
+  vs.date_of_visit
+  FROM visits vs LEFT JOIN animals a ON a.id = vs.animals_id
+  LEFT JOIN vets v 
+    ON v.id = vs.vets_id
+  WHERE 
+    v.name = 'Stephanie Mendez' AND 
+    vs.date_of_visit 
+    BETWEEN 'Apr 1, 2020' AND 'Aug 30, 2020';
+
+SELECT a.name, COUNT(*) 
+  FROM visits vs INNER JOIN animals a ON a.id = vs.animals_id
+  GROUP BY a.name
+  ORDER BY COUNT DESC
+  LIMIT 1;
+
+SELECT 
+  a.name AS animal_name, 
+  v.name AS vet_name,
+  vs.date_of_visit
+  FROM visits vs LEFT JOIN animals a ON a.id = vs.animals_id
+  LEFT JOIN vets v ON v.id = vs.vets_id
+  WHERE v.name = 'Maisy Smith'
+  ORDER BY vs.date_of_visit ASC
+  LIMIT 1;
+
+SELECT 
+  a.id AS animal_id,
+  a.name AS animal_name,
+  a.date_of_birth,
+  v.id AS vet_id,
+  v.name AS vet_name, 
+  v.age AS vet_age,
+  date_of_visit
+  FROM visits vs INNER JOIN animals a ON a.id = vs.animals_id
+  INNER JOIN vets v
+  ON v.id = vs.vets_id;
```

Comment:
- The query states `Details for MOST RECENT visit: animal information, vet information, and date of visit.` Kindly implement this.

- This is what I'm getting at the moment

![](https://i.imgur.com/SBZtNPj.png)

here 👇 is an example of how to do this.


```sql
 SELECT animals.*, vets.*, vistis.visit_date
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  ORDER BY visits.visit_date DESC
  LIMIT 1;
```